posthog/api/feature_flag.py

```diff
@@ -1083,41 +1096,21 @@ def activity(self, request: request.Request, **kwargs):
         )
         return activity_page_response(activity_page, limit, page, request)
 
-    def perform_create(self, serializer):
-        serializer.save()
-        log_activity(
-            organization_id=self.organization.id,
-            team_id=self.team_id,
-            user=serializer.context["request"].user,
-            was_impersonated=is_impersonated_session(serializer.context["request"]),
-            item_id=serializer.instance.id,
-            scope="FeatureFlag",
-            activity="created",
-            detail=Detail(name=serializer.instance.key),
-        )
-
-    def perform_update(self, serializer):
-        instance_id = serializer.instance.id
 
-        try:
-            before_update = FeatureFlag.objects.get(pk=instance_id)
-        except FeatureFlag.DoesNotExist:
-            before_update = None
-
-        serializer.save()
-
-        changes = changes_between("FeatureFlag", previous=before_update, current=serializer.instance)
-
-        log_activity(
-            organization_id=self.organization.id,
-            team_id=self.team_id,
-            user=serializer.context["request"].user,
-            was_impersonated=is_impersonated_session(serializer.context["request"]),
-            item_id=instance_id,
-            scope="FeatureFlag",
-            activity="updated",
-            detail=Detail(changes=changes, name=serializer.instance.key),
-        )
+@receiver(model_activity_signal, sender=FeatureFlag)
+def handle_feature_flag_change(sender, scope, before_update, after_update, activity, was_impersonated=False, **kwargs):
+    log_activity(
+        organization_id=after_update.team.organization_id,
+        team_id=after_update.team_id,
+        user=after_update.last_modified_by,
```

Comment:
ah... this was going to be my question but you have the acting user captured on your model 💅 

i think this is great... keeps the two concerns separate

the activity log isn't bound directly to django model so it stays as flexible as today
but folk whose django model carries all the information they need can have a simpler implementation

Reply:
> the activity log isn't bound directly to django model so it stays as flexible as today but folk whose django model carries all the information they need can have a simpler implementation

Yup! I imagine that the activity log isn't bound to a specific model, but it's probably usually triggered by a model change. So in theory you could have code that listens to multiple models and decides how to write the activity log. This'll make the listening part easier in those scenarios.